test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1496,26 +1514,60 @@ public void queryClusterStatus() throws Exception {
         // STEP 4.
         // Since there will be no epoch change as majority of servers are down, we rely on PORT_2
         // to send a MANAGEMENT_FAILURE_DETECTED after which we query the cluster status.
-        Semaphore latch = new Semaphore(1);
-        latch.acquire();
+        Semaphore latch1 = new Semaphore(1);
+        latch1.acquire();
         addClientRule(getManagementServer(SERVERS.PORT_2).getManagementAgent().getCorfuRuntime(),
                 new TestRule().matches(corfuMsg -> {
                     if (corfuMsg.getMsgType().equals(CorfuMsgType.MANAGEMENT_FAILURE_DETECTED)) {
-                        latch.release();
+                        latch1.release();
                         return true;
                     }
                     return false;
                 }).drop());
         addServerRule(SERVERS.PORT_1, new TestRule().drop().always());
-        getManagementServer(SERVERS.PORT_1).shutdown();
-        assertThat(latch.tryAcquire(PARAMETERS.TIMEOUT_LONG.toMillis(), TimeUnit.MILLISECONDS))
+        addClientRule(getManagementServer(SERVERS.PORT_1).getManagementAgent().getCorfuRuntime(),
+                new TestRule().always().drop());
+        assertThat(latch1.tryAcquire(PARAMETERS.TIMEOUT_LONG.toMillis(), TimeUnit.MILLISECONDS))
                 .isTrue();
         clusterStatus = getCorfuRuntime().getManagementView().getClusterStatus();
         nodeStatusMap = clusterStatus.getClientServerConnectivityStatusMap();
         assertThat(nodeStatusMap.get(SERVERS.ENDPOINT_0)).isEqualTo(NodeStatus.DOWN);
         assertThat(nodeStatusMap.get(SERVERS.ENDPOINT_1)).isEqualTo(NodeStatus.DOWN);
         assertThat(nodeStatusMap.get(SERVERS.ENDPOINT_2)).isEqualTo(NodeStatus.UP);
         assertThat(clusterStatus.getClusterStatus()).isEqualTo(ClusterStatus.UNAVAILABLE);
+
+
+        // STEP 5.
+        // PORT_1 is revived and PORT_0 is also revived. PORT_0 attempts to add itself back to the
+        // log unit chain.
+        Semaphore latch2 = new Semaphore(1);
```

Comment:
Maybe CountdownLatch?

Reply:
I cannot combine the latches. This is due to both the latch monitoring different messages.
A countdown latch will release the semaphore on multiple message requests of the same type which is not something we want. Although what I could have done is remove the first rule and then reuse the same latch for the second rule.